### PR TITLE
ESX.OneSync.SpawnVehicle and Command rewrite

### PR DESCRIPTION
**Pull request by TheFantomas**
Some changes on ESX.OneSync.SpawnVehicle
- If you now sitting in vehicle and use command your vehicle will be deleted
- Default vehicle is now adder
- Vehicles are displayed immediately without delay. And it doesn't bug and throw people away.
- esx:requestModel was deleted because this function does not need it... Mycroft Mycroft.

### CODE REVIEW

File: [core]/es_extended/config.lua [hunk 1]
```diff
@@ -60,7 +60,7 @@ Config.RemoveHudCommonents = {
 	[22] = false, --HUD_WEAPONS
 }
 
-Config.MaxAdminVehicles = false -- admin vehicles spawn with max vehcle settings
+Config.MaxAdminVehicles = true -- admin vehicles spawn with max vehcle settings
```
Comment (Gellipapa) [inline [core]/es_extended/config.lua]: SpawnVehMaxUpgrades i think it's better variable name.

File: [core]/es_extended/server/commands.lua [hunk 2]
```diff
@@ -23,29 +23,49 @@ end, true, {help = TranslateCap('command_setjob'), validate = true, arguments =
...
+        modTransmission = 2,
+        modSuspension = 3,
+        modArmor = true,
+        windowTint = 1
+    } or {}
+
 ESX.RegisterCommand('car', 'admin', function(xPlayer, args, showError)
-	local GameBuild = tonumber(GetConvar("sv_enforceGameBuild", 1604))
-	if not args.car then args.car = GameBuild >= 2699 and "draugur" or "prototipo" end
+	local playerPed = GetPlayerPed(xPlayer.source)
```
Comment (Gellipapa) [inline [core]/es_extended/server/commands.lua:38]: You should check if xPlayer is not nil.

File: [core]/es_extended/server/commands.lua [hunk 3]
```diff
@@ -23,29 +23,49 @@ end, true, {help = TranslateCap('command_setjob'), validate = true, arguments =
...
-		Wait(250)
-		TaskWarpPedIntoVehicle(PlayerPed, vehicle, -1)
+
+	ESX.OneSync.SpawnVehicle(args.car, playerCoords, playerHeading, upgrades, function(networkId)
+		if networkId then
+			local vehicle = NetworkGetEntityFromNetworkId(networkId)
+			for i = 1, 20 do
+				Wait(0)
+				SetPedIntoVehicle(playerPed, vehicle, -1)
+
```
Comment (Gellipapa) [inline [core]/es_extended/server/commands.lua:63]: I think if you can't adjust the ped in the vehicle you should throw any generic error message.

File: [core]/es_extended/config.lua (hunk 1 above)
Comment (TheFantomas) [reply [core]/es_extended/config.lua]: Updated

File: [core]/es_extended/server/commands.lua (hunk 3 above)
Comment (TheFantomas) [reply [core]/es_extended/server/commands.lua]: Done

File: [core]/es_extended/server/commands.lua (hunk 2 above)
Comment (TheFantomas) [reply [core]/es_extended/server/commands.lua]: Done

File: [core]/es_extended/server/commands.lua [hunk 4]
```diff
@@ -35,41 +35,45 @@ local upgrades = Config.SpawnVehMaxUpgrades and
...
-    local playerVehicle = GetVehiclePedIsIn(playerPed)
-
-	if not args.car or type(args.car) ~= 'string' then
-		args.car = 'adder'
-	end
-
-	if playerVehicle then
-		DeleteEntity(playerVehicle)
-	end
+	if xPlayer then
```
Comment (Gellipapa) [inline [core]/es_extended/server/commands.lua:38]: reverse logic is better and clean

if not xPlayer then
  return notify....
end

**Comment by Gellipapa** [comment]
Thank you good job 👍

**Review by Gellipapa — APPROVED** [review]

**Comment by Gellipapa** [comment]
> Some who know to fix this error:
> 
> [ script:es_extended] SCRIPT ERROR: @es_extended/server/onesync.lua:71: attempt to call a nil value (global 'CreateVehicleServerSetter') [ script:es_extended] > ? (@es_extended/server/onesync.lua:71) [ script:es_extended] > handler (@es_extended/server/common.lua:92)
> 
> Its come when i try to spawn a car

Hi! Thank you for report this issue. @TheFantomas did this he knows what the problem might be.

**Comment by TheFantomas** [comment]
> Some who know to fix this error:
> 
> [ script:es_extended] SCRIPT ERROR: @es_extended/server/onesync.lua:71: attempt to call a nil value (global 'CreateVehicleServerSetter') [ script:es_extended] > ? (@es_extended/server/onesync.lua:71) [ script:es_extended] > handler (@es_extended/server/common.lua:92)
> 
> Its come when i try to spawn a car

Update your server artifacts please. I guess you have some old artifacts version. Please use discord support channels please for support.

**Pull request merged**